### ISSUE TITLE
What android version does the virtual app need to be?

### ISSUE THREAD

**Issue by IlIiIlIIlIiIlIlIIIlI** [sim0n00ps/L3-Dumping#10]
It's defaulting me to Android 14 on a Pixel 6, and it does not work. I have zero errors in either CMD window, and have followed all the instructions.

Loading the video in chrome says "SOURCE_ERROR" on Pixel 7 Pro on Android 14, and on Pixel 6 on Android 14, it just simply never generates the dump keys. Bitmovin also only loads the site like 25% of the time, and Chrome randomly closes sometimes once functions are hooked.

2023-12-18 12:31:24 AM - main - 32 - INFO - Connected to Android Emulator 5554
2023-12-18 12:31:24 AM - main - 33 - INFO - Scanning all processes
2023-12-18 12:31:26 AM - main - 39 - INFO - Functions hooked, now open the DRM stream test on Bitmovin from your Android device! https://bitmovin.com/demos/drm

Any idea's? Been working at this for like 4 hours, and have hit a brick wall. Do I even really need to do this, the OFD seems to work without it, or is it missing DRM content when it runs?

**Comment by IlIiIlIIlIiIlIlIIIlI** [comment]
Nevermind, FINALLY figured it out.

Android version MUST be Version 9 (Pie), and it MUST be x86_64 not x86 which is recommended. I used the Pixel 6.

Secondly you must download the x86_64 version of frida server, and you must have the same version of frida installed via shell, which you can check with "pip show frida", version number must match server.

**I finally got the keys to export, but I still have one question**, which is my keys are not named what you referenced. I see in the thread one person says just to rename them, but will that work?

Mine are named client_id and private_key.pem

So just rename like below?
client_id.bin -> device_client_id_blob
private_key.pem -> device_private_key 

OFD seems to detect them after renaming as "good" / green, but not sure if they actually work or not?

**Comment by sim0n00ps** [comment]
> Nevermind, FINALLY figured it out.
> 
> Android version MUST be Version 9 (Pie), and it MUST be x86_64 not x86 which is recommended. I used the Pixel 6.
> 
> Secondly you must download the x86_64 version of frida server, and you must have the same version of frida installed via shell, which you can check with "pip show frida", version number must match server.
> 
> **I finally got the keys to export, but I still have one question**, which is my keys are not named what you referenced. I see in the thread one person says just to rename them, but will that work?
> 
> Mine are named client_id and private_key.pem
> 
> So just rename like below? client_id.bin -> device_client_id_blob private_key.pem -> device_private_key
> 
> OFD seems to detect them after renaming as "good" / green, but not sure if they actually work or not?

Yes the tool will not export them with the correct names, you need to rename them how you've said in your comment, then they should work correctly
